gnwmanager/gnw.py

```diff
@@ -504,6 +513,96 @@ def _flash_ext(
 
         self.wait_for_all_contexts_complete()
 
+    def _sd_write_file_chunk(
+        self,
+        path: str,
+        block: int,
+        total_blocks: int,
+        data: bytes,
+        blocking: bool = True,
+        compress: bool = True,
+    ) -> None:
+        """Low-level write data to fat filesystem.
+
+        Limited to RAM constraints (i.e. <256KB writes).
+
+        Parameters
+        ----------
+        path: str
+            Path of file to write
+        size: int
+            Number of bytes to write.
+        blocking: bool
+            Wait for action to be complete.
+        """
+        log.debug(f"gnw._sd_write_file_chunk: {path=} {len(data)=} {blocking=} {compress=}")
+
+        if not path:
+            return
```

Comment:
I think this should `raise ValueError("Destination SD path cannot be empty.")` instead